Framework/Geometry/src/Instrument/InstrumentVisitor.cpp

```diff
@@ -343,6 +345,12 @@ InstrumentVisitor::makeWrappers() const {
   return {std::move(compInfoWrapper), std::move(detInfoWrapper)};
 }
 
+bool InstrumentVisitor::componentIsVisible(
+    Mantid::Geometry::IComponent *componentID) {
```

Comment:
Not bad. Why didn't you add an `isVisible` method to `IComponent` and go through that? Any reason?